src/main/java/seedu/bankwithus/AccountList.java

```diff
@@ -195,21 +211,35 @@ public void withdrawMoney(String withdrawAmountString) throws NumberFormatExcept
             throw new NegativeAmountException();
         }
         float currentBalance = Float.parseFloat(getMainAccount().getAccountBalance());
+        float expectedBal = currentBalance - withdrawAmount;
+        LocalDate tdy = LocalDate.now();
+        LocalDate tdyDate = handleDate(tdy);
         if (currentBalance < withdrawAmount) {
             throw new InsufficientBalanceException();
+        } else if(isFailsSaveGoal(expectedBal, tdyDate)) {
+            ui.failToMeetSaveGoal();
+            handleProceed(withdrawAmount, currentBalance);
         } else {
-            getMainAccount().subtractBalance(currentBalance,withdrawAmount);
+            getMainAccount( ).subtractBalance(currentBalance,withdrawAmount);
+            ui.showWithdrawMessage();
+        }
+    }
+
+    public void findAccount(String name, Account acc) {
+        if (acc.getAccountName().contains(name)) {
+            accounts.remove(acc);
+            ui.showAccountDeleted(name);
+            if(accounts.size() < 1) {
+                createNewAccount();
+            }
```

Comment:
Why is the name of the function find account when it also deletes the account? It is better to make a pure findAccount function, and then do the removal of the account in the deleteAccount function